bind.go

```diff
@@ -60,9 +60,14 @@ func (l *Conn) SimpleBind(simpleBindRequest *SimpleBindRequest) (*SimpleBindResu
 	}
 	defer l.finishMessage(messageID)
 
-	packet = <-channel
-	if packet == nil {
-		return nil, NewError(ErrorNetwork, errors.New("ldap: could not retrieve response"))
+	packetResponse, ok := <-channel
+	packet, err = packetResponse.ReadPacket()
```

Comment:
I take it back, hadn't gotten to the nil receiver impl below... that's nonstandard enough that I think you should always check `ok` before doing anything with `packetResponse`